QtCollider/QcApplication.cpp

```diff
@@ -104,6 +104,29 @@ QcApplication::~QcApplication() {
     _mutex.unlock();
 }
 
+void QcApplication::createMenu() {
+    _mainMenu = QSharedPointer<QMenuBar>(new QMenuBar(0));
+
+#ifdef Q_OS_MAC
+    // macOS registers cmd+q on menu bars by default. Here we register a handler for cmd+q that does nothing
+    // and we disable the Quit menu by default
+    QMenu* menu;
+    QAction* action;
+    action = new QAction(tr("Catch CMD+Q on MacOS X"));
+    action->setMenuRole(QAction::QuitRole);
+    QObject::connect(action, SIGNAL(triggered()), this, SLOT(onQuit()));
+    menu = new QMenu(tr("&File"));
+    menu->addAction(action);
+    _mainMenu->addMenu(menu);
+    action->setEnabled(false);
+#endif
+}
+
+void QcApplication::onQuit() {
+    printf("[QcApplication::onQuit] CMD+Q was caught on the interpreter. This is weird, it should not happen. \
```

Comment:
These are only wording suggestions, feel free to disregard:
`CMD+Q was caught by the Interpreter.`

Reply:
That's better indeed, thanks !